worker/peergrouper/worker.go

```diff
@@ -438,14 +438,14 @@ func (w *pgWorker) getMongoSpace(addrs [][]network.Address) (network.SpaceName,
 			}
 			logger.Warningf("couldn't find a space containing all peer group machines")
 			return unset, nil
-		} else {
-			spaceName, err := w.config.State.SetOrGetMongoSpaceName(spaceStats.LargestSpace)
-			if err != nil {
-				return unset, errors.Annotate(err, "error setting/getting Mongo space")
-			}
-			return spaceName, nil
 		}
 
+		spaceName, err := w.config.State.SetOrGetMongoSpaceName(spaceStats.LargestSpace)
```

Comment:
I don't quite understand why this is safe to pull out of the else clause.
I guess the if() portion returns, which means that putting it in 'else' is the same as putting it inline?
Is it more obvious to humans if we leave it in the else ?

I guess it is fairly clear that if the precondition is false then we return, thus we can pull it out.

Reply:
This needn't have accompanied the PR. I did this in a first pass, anticipating working with this file before ending up changing the MAAS provider.

In any case, I can reinstate the block if you think it clearer the other way. Perhaps it is pedantic, but when reading code that has a some semantic that one could otherwise omit, I hit a little speed bump in my reading flow.